### PR TITLE
KAFKA-10028: Minor fixes to describeFeatures and updateFeatures apis

### PR DESCRIPTION
**Pull request by kowshik**
In this PR, I have addressed the review comments from @chia7712 in https://github.com/apache/kafka/pull/9001 which were provided after https://github.com/apache/kafka/pull/9001 was merged. The changes are made mainly to `KafkaAdminClient`:
1. Improve error message in `updateFeatures` api when feature name is empty.
2. Propagate top-level error message in `updateFeatures` api.
3. Add an empty-parameter variety for `describeFeatures` api.
4. Minor documentation updates to `@param` and `@return` to make these resemble other apis.

**Test plan:**
Relying on existing unit and integration tests.

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by kowshik** [comment]
@chia7712 thanks for the review! would you be able to please help merge this PR into trunk when CI passes?

**Comment by chia7712** [comment]
I can't merge it since I'm not a Kafka committer 😅 

@junrao Could you please take a look?

**Review by junrao — APPROVED** [review]
@kowshik : Thanks for the PR. LGTM

**Comment by junrao** [comment]
In this PR, I have addressed the review comments from @chia7712 in #9001 which were provided after #9001 was merged. The changes are made mainly to KafkaAdminClient:

Improve error message in updateFeatures api when feature name is empty.
Propagate top-level error message in updateFeatures api.
Add an empty-parameter variety for describeFeatures api.
Minor documentation updates to @param and @return to make these resemble other apis.

Reviewers: Chia-Ping Tsai <chia7712@gmail.com>, Jun Rao <junrao@gmail.com>

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:1385]: It appears we forgot to add an overload for updateFeatures. @kowshik @junrao WDYT? Is it worth adding consistency for now?

Comment (junrao) [reply clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:1385]: Thanks for finding this. Yes, we can add an overload w/o options for consistency. It would be useful to update the original KIP and the discussion thread with the change.

Comment (brandboat) [reply clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:1385]: Thanks @chia7712, @junrao for raising up the issue, I've fixed this in PR: https://github.com/apache/kafka/pull/21380

discussion thread update: https://lists.apache.org/thread/4vwgj3f7l9nkwq0rpfmxq98zwo48pxqj
KIP is also updated: https://cwiki.apache.org/confluence/display/KAFKA/KIP-584%3A+Versioning+scheme+for+features

**Pull request merged**
